packages/serverpod/lib/src/database/database_query.dart

```diff
@@ -41,27 +44,72 @@ class SelectQueryBuilder {
       orderBy: _orderBy,
       where: _where,
     );
-
     var selectColumns = [..._fields, ..._gatherIncludeColumns(_include)];
 
     var subQueries = _buildSubQueries(orderBy: _orderBy);
     var select = _buildSelectStatement(selectColumns);
     var join =
         _buildJoinQuery(where: _where, orderBy: _orderBy, include: _include);
     var groupBy = _buildGroupByQuery(selectColumns, orderBy: _orderBy);
-    var where = _buildWhereQuery(where: _where);
-    var orderBy = _buildOrderByQuery(orderBy: _orderBy);
+    var where = _buildWhereQuery(
+      where: _where,
+      listQueryAdditions: _listQueryAdditions,
+    );
 
     var query = '';
-    if (subQueries != null) query += '$subQueries ';
     query += 'SELECT $select';
     query += ' FROM "${_table.tableName}"';
     if (join != null) query += ' $join';
     if (where != null) query += ' WHERE $where';
-    if (groupBy != null) query += ' GROUP BY $groupBy';
-    if (orderBy != null) query += ' ORDER BY $orderBy';
-    if (_limit != null) query += ' LIMIT $_limit';
-    if (_offset != null) query += ' OFFSET $_offset';
+
+    if (groupBy != null) query += ' $groupBy';
+    if (_orderBy != null) {
+      query +=
+          ' ORDER BY ${_orderBy?.map((order) => order.toString()).join(', ')}';
+    }
+
+    var limit = _limit;
+
+    if (_listQueryAdditions != null && limit != null) {
+      return _wrapListQueryWithLimit(
+        query,
+        subQueries,
+        limit: limit,
+        offset: _offset,
+      );
+    } else {
+      if (limit != null) query += ' LIMIT $limit';
+      if (_offset != null) query += ' OFFSET $_offset';
+
+      if (subQueries != null) query = '$subQueries $query';
+      return query;
+    }
+  }
+
+  String _wrapListQueryWithLimit(
+    String baseQuery,
+    String? subQueries, {
+    required int limit,
+    required int? offset,
+  }) {
+    var wrappedBaseQueryAlias = '_base_query_sorting_and_ordering';
+    var partitionedQueryAlias = '_partitioned_list_by_parent_id';
+
+    var index = offset ?? 0;
+    var start = index + 1;
+    var end = limit + index;
+
+    var relationalFieldName =
+        unescape(_listQueryAdditions!.relationalFieldName);
```

Comment:
Question: any reason for not saving it unescaped directly?

Reply:
Because we need to use it escaped in some scenarios. This was an easy solution :)

Reply:
modified!